app/models/mixins/aggregation_mixin/methods.rb

```diff
@@ -39,13 +39,10 @@ def all_storages
     end
 
     def aggregate_hardware(from, field, targets = nil)
-      from      = from.to_s.singularize
-      select    = field == :aggregate_cpu_speed ? "cpu_total_cores, cpu_speed" : field
-      targets ||= send("all_#{from}_ids")
-      targets   = targets.collect(&:id) unless targets.first.kind_of?(Integer)
-      hdws      = Hardware.where("#{from}_id" => targets).select(select)
-
-      hdws.inject(0) { |t, hdw| t + hdw.send(field).to_i }
+      col_name = "from.to_s.singularize}_id"
```

Comment:
I have considered introducing an association for this. (e.g.: `host_hardware`) and just running the `sum` through this and being 100% AR